File: dev-scripts/migrate_to_supabase.mjs
import { createClient } from '@supabase/supabase-js';
import { readFileSync } from 'fs';

const SUPABASE_URL = 'https://kvzltutlfuvyzigtmoqw.supabase.co';
const SUPABASE_KEY = 'sb_secret_pdlPG6W2OcEr2wWxeS2AGQ_WyvkE7hK';

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);

async function migrate() {
    // --- POEMS ---
    console.log('📖 Loading poems.json...');
    const poems = JSON.parse(readFileSync('src/data/poems.json', 'utf-8'));

    const poemRows = poems.map(p => ({
        id: String(p.id),
        title: p.title || '',
        date: p.date || null,
        style: p.style || '',
        theme: p.theme || '',
        meter: p.meter || '',
        dedication: p.dedication || '',
        classification: p.classification || null,
        is_pinned: p.isPinned || false,
        pin_expires_at: p.pinExpiresAt || null,
        urai: p.urai || '',
        notes: p.notes || '',
        variants: p.variants || []
    }));

    console.log(`   Uploading ${poemRows.length} poems...`);
    const { data: pData, error: pErr } = await supabase.from('poems').upsert(poemRows);
    if (pErr) {
        console.error('   ❌ Poems error:', pErr.message);
    } else {
        console.log(`   ✅ ${poemRows.length} poems uploaded!`);
    }

    // --- QUOTES ---
    console.log('💬 Loading quotes.json...');
    const quotes = JSON.parse(readFileSync('src/data/quotes.json', 'utf-8'));

    const quoteRows = quotes.map(q => ({
        id: String(q.id),
        tag: q.tag || '',
        date: q.date || null,
        is_pinned: q.isPinned || false,
        pin_expires_at: q.pinExpiresAt || null,
        urai: q.urai || '',
        notes: q.notes || '',
        variants: q.variants || []
    }));

    console.log(`   Uploading ${quoteRows.length} quotes...`);
    const { data: qData, error: qErr } = await supabase.from('quotes').upsert(quoteRows);
    if (qErr) {
        console.error('   ❌ Quotes error:', qErr.message);
    } else {
        console.log(`   ✅ ${quoteRows.length} quotes uploaded!`);
    }

    // --- VERIFY ---
    console.log('\n🔍 Verifying...');
    const { data: allPoems } = await supabase.from('poems').select('id,title,theme,style');
    console.log(`   Poems in Supabase: ${allPoems?.length || 0}`);
    allPoems?.forEach(p => console.log(`     [${(p.theme || '').padEnd(12)}] [${(p.style || '').padEnd(16)}] ${p.title}`));

    const { data: allQuotes } = await supabase.from('quotes').select('id,tag,date');
    console.log(`   Quotes in Supabase: ${allQuotes?.length || 0}`);
    allQuotes?.forEach(q => console.log(`     [${(q.tag || '').padEnd(15)}] [${(q.date || '').toString().padEnd(20)}]`));

    console.log('\n🎉 Migration complete!');
}

migrate().catch(err => {
    console.error('Fatal error:', err);
    process.exit(1);
});

File: dev-scripts/migrate_to_supabase.py
import json
import requests
import sys

# Supabase config
SUPABASE_URL = "https://kvzltutlfuvyzigtmoqw.supabase.co"
SUPABASE_KEY = "sb_secret_pdlPG6W2OcEr2wWxeS2AGQ_WyvkE7hK"

HEADERS = {
    "apikey": SUPABASE_KEY,
    "Authorization": f"Bearer {SUPABASE_KEY}",
    "Content-Type": "application/json",
    "Prefer": "return=representation"
}

def upload_poems():
    print("📖 Loading poems.json...")
    with open("src/data/poems.json", "r", encoding="utf-8") as f:
        poems = json.load(f)
    
    rows = []
    for p in poems:
        row = {
            "id": str(p["id"]),
            "title": p.get("title", ""),
            "date": p.get("date"),
            "style": p.get("style", ""),
            "theme": p.get("theme", ""),
            "meter": p.get("meter", ""),
            "dedication": p.get("dedication", ""),
            "classification": p.get("classification"),
            "is_pinned": p.get("isPinned", False),
            "pin_expires_at": p.get("pinExpiresAt"),
            "urai": p.get("urai", ""),
            "notes": p.get("notes", ""),
            "variants": p.get("variants", [])
        }
        rows.append(row)
    
    print(f"   Uploading {len(rows)} poems...")
    resp = requests.post(
        f"{SUPABASE_URL}/rest/v1/poems",
        headers=HEADERS,
        json=rows
    )
    
    if resp.status_code in (200, 201):
        print(f"   ✅ {len(rows)} poems uploaded successfully!")
        return True
    else:
        print(f"   ❌ Error: {resp.status_code} — {resp.text}")
        return False

def upload_quotes():
    print("💬 Loading quotes.json...")
    with open("src/data/quotes.json", "r", encoding="utf-8") as f:
        quotes = json.load(f)
    
    rows = []
    for q in quotes:
        row = {
            "id": str(q["id"]),
            "tag": q.get("tag", ""),
            "date": q.get("date"),
            "is_pinned": q.get("isPinned", False),
            "pin_expires_at": q.get("pinExpiresAt"),
            "urai": q.get("urai", ""),
            "notes": q.get("notes", ""),
            "variants": q.get("variants", [])
        }
        rows.append(row)
    
    print(f"   Uploading {len(rows)} quotes...")
    resp = requests.post(
        f"{SUPABASE_URL}/rest/v1/quotes",
        headers=HEADERS,
        json=rows
    )
    
    if resp.status_code in (200, 201):
        print(f"   ✅ {len(rows)} quotes uploaded successfully!")
        return True
    else:
        print(f"   ❌ Error: {resp.status_code} — {resp.text}")
        return False

def verify():
    print("\n🔍 Verifying...")
    
    resp = requests.get(
        f"{SUPABASE_URL}/rest/v1/poems?select=id,title,theme,style",
        headers=HEADERS
    )
    poems = resp.json()
    print(f"   Poems in Supabase: {len(poems)}")
    for p in poems:
        print(f"     [{p.get('theme',''):12s}] [{p.get('style',''):16s}] {p.get('title','')}")
    
    resp = requests.get(
        f"{SUPABASE_URL}/rest/v1/quotes?select=id,tag,date",
        headers=HEADERS
    )
    quotes = resp.json()
    print(f"   Quotes in Supabase: {len(quotes)}")
    for q in quotes:
        print(f"     [{q.get('tag',''):15s}] [{q.get('date',''):20s}]")

if __name__ == "__main__":
    ok1 = upload_poems()
    ok2 = upload_quotes()
    if ok1 and ok2:
        verify()
        print("\n🎉 Migration complete!")
    else:
        print("\n⚠️  Some uploads failed. Check the errors above.")
        sys.exit(1)
